packages/core/js-client/src/connection/RelayConnection.ts

```diff
@@ -178,30 +180,39 @@ export class RelayConnection implements IConnection {
 
         await this.lib2p2Peer.handle(
             [PROTOCOL_NAME],
-            async ({ connection, stream }) => {
-                pipe(
-                    stream.source,
-                    // @ts-ignore
-                    decode(),
-                    // @ts-ignore
-                    (source) => map(source, (buf) => toString(buf.subarray())),
-                    async (source) => {
-                        try {
-                            for await (const msg of source) {
-                                try {
-                                    const particle = Particle.fromString(msg);
-                                    log.trace('got particle from stream with id %s and particle id %s', stream.id, particle.id);
+            async ({ connection, stream }) => pipe(
+                stream.source,
+                decode(),
+                (source) => map(source, (buf) => toString(buf.subarray())),
+                async (source) => {
+                    try {
+                        for await (const msg of source) {
+                            try {
+                                const particle = Particle.fromString(msg);
+                                const initPeerId = peerIdFromString(particle.initPeerId);
+                                
+                                if (initPeerId.type !== 'Ed25519') {
+                                    log.error('unsupported peer id format in particle with id = %s. expected: %s. given: %s', particle.id, 'Ed25519', initPeerId.type);
+                                    return;
+                                }
+                                
+                                const publicKey = unmarshalPublicKey(initPeerId.publicKey);
+                                log.trace('got particle from stream with id %s and particle id %s', stream.id, particle.id);
+                                const isVerified = await verifySignature(particle, publicKey);
+                                if (isVerified) {
                                     this.particleSource.next(particle);
-                                } catch (e) {
-                                    log.error('error on handling a new incoming message: %j', e);
+                                } else {
+                                    log.trace('particle signature doesn\'t match with the message. rejecting particle with id: %s', particle.id);
```

Comment:
I don't think we have anything called "message".
```suggestion
                                    log.trace('particle signature is incorrect. rejecting particle with id: %s', particle.id);
```

Reply:
Fixed